src/Microsoft.VisualStudio.Services.Agent/Util/PipelineTasksUtil.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Agent.Sdk;
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    public static class PipelineTasksUtil
+    {
+        public static readonly Dictionary<string, WellKnownScriptShell> ScriptShellsPerTaskNames = new Dictionary<string, WellKnownScriptShell>(StringComparer.OrdinalIgnoreCase)
+        {
+            ["PowerShell"] = WellKnownScriptShell.PowerShell,
+            ["Bash"] = WellKnownScriptShell.Bash
+        };
+
+        public static WellKnownScriptShell GetShellByTaskName(string taskName, ExecutionTargetInfo target)
+        {
+            if (target is ContainerInfo targetContainer)
+            {
+                return GetShellByTaskName(taskName, targetContainer);
+            }
+
+            return GetShellByTaskName(taskName);
+        }
+
+        public static WellKnownScriptShell GetShellByTaskName(string taskName, ContainerInfo targetContainer)
+        {
+            ArgUtil.NotNull(targetContainer, nameof(targetContainer));
+
+            // CmdLine works like cmd on Windows and bash on Linux or MacOs.
```

Comment:
>CmdLine works like cmd on Windows and bash on Linux or MacOs [](http://example.com/codeflow?start=15&length=60)

[suggestion not blocking]

"CmdLine works like cmd on Windows and bash on Linux or MacOs"

I'm not sure how this assumption is used, but it's not correct -- bash (minggw) can be installed and used on windows, for example using Git installer.  I've seen an internal team use the bash task on Windows (for portable scripts between linux/windows)

Reply:
Yes, but I mean that "CmdLine" task works as cmd in case it's running on Windows, see:
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/CmdLineV2/task.json#L66-L71

This behavior is typical only for the CmdLine task

For Linux and macOS "CmdLine" task will be executed as `cmdline.js` target where Bash is used:
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/CmdLineV2/task.json#L72-L79